src/deadline/client/ui/dialogs/submit_job_to_deadline_dialog.py

```diff
@@ -82,6 +83,8 @@ def __init__(
         auto_detected_attachments: AssetReferences,
         attachments: AssetReferences,
         on_create_job_bundle_callback,
+        on_ui_callback=None,
+        on_post_submit_callback=None,
```

Comment:
I know we started down this road of callbacks being passed in here, but I wonder if these two cases are better served via a plugin structure like what @epmog alluded to in https://github.com/aws-deadline/deadline-cloud/issues/308#issuecomment-2103569689 ?

The repo + blogpost https://github.com/wax911/plugin-architecture looks like a decent high level way to think about plugins - with discovery and registration being the key concepts to think about first. With callbacks via the parameter here, it is the code that is using the deadline-cloud library that has to perform any discovery of plugins available.

For `on_post_submit_callback`, does this meet the goals of the callback? If we want to let someone write code that runs on every single job submission, this won't work, because every downstream dependency would need to implement specific code to support that, and other ways of submitting jobs like the CLI `deadline bundle submit` don't use these GUI code flows at all.

If we have a "post submit" action that a plugin can register, and within the deadline-cloud library every path to submitting a job runs the plugins (probably with an explicit opt-out `--no-plugins` option or similar to help with testing whether a plugin is responsible for an issue or not), maybe that's closer to the extensibility we want?